cmd/vault.go

```diff
@@ -332,19 +336,20 @@ func secretSearchCmd() *cobra.Command {
 
 	flags := cmd.Flags()
 	flags.IntVar(&options.offset, "offset", 0, "where to start fetching the items")
-	flags.IntVar(&options.limit, "limit", 50, "number of items to return")
+	flags.IntVar(&options.limit, "limit", 100, "number of items to return")
 	flags.StringVar(&options.sortdir, "sortdir", "", "sort direction, ASC or DESC (default ASC)")
 	flags.StringVar(&options.sortkey, "sortkey", "", "sort object by name, updated, or created.")
 	flags.StringVar(&options.keywords, "keywords", "", "comma or space-separated string to search in secret's names")
+	flags.StringVar(&options.filter, "filter", "", "Defines what type of secrets to search for. personal, shared, readable, writeable") //shared not working
```

Comment:
I think that if you for example put a wrong "filter" the api will tell you in the response that the filter is wrong.
Do you want me to verify "sortdir, sortkey, keywords and filter" before doing the API call?

Reply:
Yes, it is best to forget what the backend might do (because you should not look from the backend insider perspective but rather from the API specifications consumer - like the web frontend). The web frontend does user input validation and tries to send only valid params to the backend. So should the go-client. 

Going back to the specifics in the question, sortdir uses string enums like ASC/DESC and filter uses string enums namely personal, shared, readable, writeable. These should be validated. Sortdir should be normalised to uppercase before validation. Similarly, filter should be normalised to lowercase.